reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,70 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
+  if not filepath or #filepath < 1 then
+    return ""
+  end
+
+  local root, _, ext = filepath:match("([^%.]*)(%.?([^\\/%.]*))")
+
+  if not root or #root < 1 then
+    return ""
+  end
+
+  if #ext > 0 then
+    return filepath
+  end
+
+  return filepath .. '.' .. extension
+end
+
+-- Returns the given path if a full path, otherwise returns given path
+-- relative to the REAPER project resource directory.
+PathUtil.get_real_path = function(path_arg)
+  if PathUtil._is_full_path(path_arg) then
+    return path_arg
+  end
+
+  local project_path = reaper.GetProjectPath()
+  return project_path .. PathUtil._path_separator() .. path_arg
+end
+
+-- Returns the command to reveal the given path in the OS file explorer.
+-- Works for Windows and macOS. To quote a wise philosopher, "Linux is a mess."
+PathUtil.get_reveal_command = function(path_arg)
+  if EnvUtil.is_windows() then
+    return '%SystemRoot%\\explorer.exe /select,"' .. path_arg .. '"'
```

Comment:
I don't think a double backslash makes sense here. For me, `%SystemRoot%` is "C:\WINDOWS"

Reply:
can you elaborate? based on how i'm reading that example the output is `C:\WINDOWSexplorer.exe` 🤔

Reply:
Sorry, I forgot backslashes need to be escaped in strings 👶